onnxruntime/core/optimizer/nchwc_transformer.cc

```diff
@@ -0,0 +1,742 @@
+    TransformPool(node);
+  } else if (node.GetInputEdgesCount() == 0 && node.InputDefs().size() != 0) {
+    // The following transforms only run when the input edge count has already
+    // been decremented to zero by earlier transforms. This is a hint that the
+    // node may already have all inputs converted to NCHWc format and is not
+    // needed for correct operation. This avoids doing extra string checks for
+    // nodes unrelated to this transformer.
+    if (graph_utils::IsSupportedOptypeVersionAndDomain(node, "Add", {7}) ||
+        graph_utils::IsSupportedOptypeVersionAndDomain(node, "Sum", {8})) {
+      TransformAdd(node);
+    } else if (graph_utils::IsSupportedOptypeVersionAndDomain(node, "Concat", {4})) {
+      TransformConcat(node);
+    } else if (graph_utils::IsSupportedOptypeVersionAndDomain(node, "Relu", {6})) {
+      TransformActivation(node);
+    } else if (graph_utils::IsSupportedOptypeVersionAndDomain(node, "Clip", {6})) {
+      TransformElementwise(node);
+    }
+  }
+
+  // The node may not match any of the checks above or may not have been
+  // transformed for other reasons such as unsupported attributes or alignment.
+  // However, the node may still use an input that has been produced by a NCHWc
+  // node. Finalize() walks through the list of NCHWc outputs and inserts the
+  // needed reorder operations to ensure that these inputs remain in NCHW
+  // format.
+}
+
+void NchwcTransformerImpl::Finalize(bool& modified) {
+  // Create ReorderOutput nodes for any NCHWc outputs that still have uses with
+  // the original tensor format.
+  for (auto& nchwc_output : nchwc_args_) {
+    if (nchwc_output.second->remaining_original_uses_ > 0) {
+      auto* output_original_arg = nchwc_output.first;
+      auto* output_nchwc_arg = nchwc_output.second->nchwc_arg_;
+      Node& reorder_output_node = graph_.AddNode(graph_.GenerateNodeName("ReorderOutput"),
+                                                 "ReorderOutput",
+                                                 "ReorderOutput",
+                                                 {output_nchwc_arg},
+                                                 {output_original_arg},
+                                                 nullptr,
+                                                 kMSNchwcDomain);
+      reorder_output_node.AddAttribute("channels", nchwc_output.second->channels_);
+      reorder_output_node.SetExecutionProviderType(kCpuExecutionProvider);
+    }
+  }
+
+  for (auto index : removed_nodes_) {
+    auto& node = *graph_.GetNode(index);
+    std::vector<Node::EdgeEnd> input_edges;
+    input_edges.reserve(node.GetInputEdgesCount());
+    for (auto it = node.InputEdgesBegin(); it != node.InputEdgesEnd(); ++it) {
+      input_edges.push_back(*it);
+    }
+    for (auto& edge : input_edges) {
+      graph_.RemoveEdge(edge.GetNode().Index(), node.Index(), edge.GetSrcArgIndex(), edge.GetDstArgIndex());
+    }
+
+    graph_.RemoveNode(index);
```

Comment:
the Graph::RemoveNode() will remove the input edges for you, so you don't need to do it manually.

Reply:
Oops, thanks, fixed.